sqlite/ext/comdb2/systables.c

```diff
@@ -80,6 +81,35 @@ static int systblSystabsDisconnect(sqlite3_vtab *pVtab)
     return SQLITE_OK;
 }
 
+/* Returns the first module in the list that should be visible. */
+static HashElem *getFirstVisibleModule(systbl_systabs_cursor *pCur)
```

Comment:
Can `getFirstVisibleModule` and `getNextVisibleModule` be combined together? If `pCur->current` is `NULL`, iterate the hashtable from the very beginning. Otherwise start from `pCur->current`. For instance,

```javascript
func getNextVisibleModule
    current = pCur->current;
    while (current = (current) ? sqliteHashNext(current) : sqliteHashFirst(pCur->tabs))
        /* Check for the HIDDEN flag */
    end
    return current
```

Reply:
With ezsystables API, we don't need them anymore.

Reply:
Right, because you get all rows in one shot with ezsystables interface.